core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -141,6 +141,7 @@ private[spark] object SparkUI {
   val DEFAULT_POOL_NAME = "default"
   val DEFAULT_RETAINED_STAGES = 1000
   val DEFAULT_RETAINED_JOBS = 1000
+  val DEFAULT_RETAINED_TASKS = 10000
```

Comment:
do we have an idea of when people are hitting this?  Personally I find 10000 small also, I was thinking more like 50000 or 100000 and users can limit further if they have the history server configured to use small amounts of memory.